src/tablecloth/time/api/adjust_frequency.clj

```diff
@@ -0,0 +1,49 @@
+(ns tablecloth.time.api.adjust-frequency
+  (:require [tech.v3.datatype :refer [emap]]
+            [tablecloth.time.utils.indexing :refer [get-index-column-or-error]]
+            [tablecloth.time.utils.datatypes :refer [get-datatype]]
+            [tablecloth.api :as tablecloth]))
+
+(defn adjust-frequency
+  "Adjusts the frequency of the time index column by applying the
+  `converter` function to the values in the time index. Returns a
+  grouped dataset that can be used with `tablecloth.api.aggregate`,
+  for example.
+
+  
+  Options are:
+
+  - categories - other columns (in addition to the time index) by
+                 that should be grouped together at the specified
+                 interval.
```

Comment:
@rsiddharthan I struggled a bit with this description. Does this make sense?

Reply:
I am having second thoughts on it being only categories. Maybe a a more generic name :columns or :include-columns?

Reply:
Word... will think on it a bit more.

Reply:
@rsiddharthan I thought about this a bit more. I also looked at python usage. They let you do something like `dataframe.group_by(['B']).resample('M')`. We could encourage that eventually to, but I think for the moment tablecloth's `group-by` is not composable. I think we ran into this before actually. So I think I will open an issue around this soon. But for the moment I took your suggestion of changing it to `include-columns`.